legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/BitemporalDeltaPlanner.java

```diff
@@ -150,7 +152,13 @@ enum DeleteMode
         this.dataFields.removeIf(field -> field.fieldName().equals(ingestMode.digestField().orElseThrow(IllegalStateException::new)));
         this.dataFields.removeIf(field -> field.fieldName().equals(sourceValidDatetimeFrom.fieldName()));
 
-        this.primaryKeys.removeIf(fieldName -> fieldName.equals(sourceValidDatetimeFrom.fieldName()));
+        boolean preserveSpecifiedField = (validityDerivation instanceof SourceSpecifiesFromDateTime && ((SourceSpecifiesFromDateTime) validityDerivation).preserveSpecifiedField().orElse(false))
+            || (validityDerivation instanceof SourceSpecifiesFromAndThruDateTime && ((SourceSpecifiesFromAndThruDateTime) validityDerivation).preserveSpecifiedField().orElse(false));
```

Comment:
We can expose a method in the parent interface (ValidityDerivation) that returns a Boolean. I see this is used very often